app/src/main/java/com/evelynhasama/bucketbuds/BucketsFragment.java

```diff
@@ -0,0 +1,159 @@
+    public void onCreate(Bundle savedInstanceState) {
+        setHasOptionsMenu(true);
+        super.onCreate(savedInstanceState);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        // Inflate the layout for this fragment
+        view = inflater.inflate(R.layout.fragment_buckets, container, false);
+        rvBuckets = view.findViewById(R.id.rvBucketsFB);
+        btnAddBucket = view.findViewById(R.id.btnAddBucketFB);
+        btnSortFilter = view.findViewById(R.id.btnSortFB);
+
+        // set default sorting/filtering
+        selectedFilterID = R.id.rbAll;
+        selectedSortID = R.id.rbModified;
+
+        btnAddBucket.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                FragmentActivity activity = getActivity();
+                Fragment myFragment = new CreateBucketFragment();
+                activity.getSupportFragmentManager().beginTransaction().replace(R.id.flContainer, myFragment).addToBackStack(null).commit();
+            }
+        });
+
+        btnSortFilter.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                showSortFilterDialog();
+            }
+        });
+
+        rvBuckets.setLayoutManager(new GridLayoutManager(getContext(), 3));
+
+        bucketLists = new ArrayList<>();
+        adapter = new BucketListAdapter(getContext(), bucketLists, selectedFilterID);
+        rvBuckets.setAdapter(adapter);
+
+        getBuckets();
+
+        return view;
+    }
+
+    public void getBuckets(){
+
+        userPub = User.getCurrentUser().getUserPubQuery();
+        FindCallback<BucketList> bucketListFindCallback = new FindCallback<BucketList>() {
+            @Override
+            public void done(List<BucketList> objects, ParseException e) {
+                if (e == null) {
+                    adapter.clear();
+                    bucketLists.clear();
+                    bucketLists.addAll(objects);
+                    adapter.notifyDataSetChanged();
+                } else {
+                    Log.d(TAG, "error adding bucket lists", e);
+                }
+            }
+        };
+        switch (selectedSortID) {
+            case R.id.rbAlphabetical:
+                userPub.getBucketsRelation().getQuery().addAscendingOrder(BucketList.KEY_NAME).findInBackground(bucketListFindCallback);
+                break;
+            case R.id.rbCreated:
+                userPub.getBucketsRelation().getQuery().addDescendingOrder(BucketList.KEY_BUCKET_CREATED).findInBackground(bucketListFindCallback);
+                break;
+            default:
+            case R.id.rbModified:
+                userPub.getBucketsRelation().getQuery().addDescendingOrder(BucketList.KEY_BUCKET_UPDATED).findInBackground(bucketListFindCallback);
+                break;
```

Comment:
nit: Code dedup opportunity.

Reply:
yeah I thought about this one when I was writing it, but wasn't sure how to implement since one is ascendingOrder and the other two are descendingOrder. any ideas?

Reply:
```
ParseQuery query = userPub.getBucketsRelation().getQuery();
switch(..) {
  case R.id.rbAlphabetical:
    query = query.addAscendingOrder(BucketList.KEY_NAME);
.....
}
query.findInBackground(bucketListFindCallback);
```